### PR TITLE
Add and document `et cleanup`.

### PR DESCRIPTION
**Pull request by matanlurey**
Closes https://github.com/flutter/flutter/issues/157945.

```sh
% et cleanup --dry-run
Checking /Users/matanl/Developer/engine/src/out...
The following directories were accessed later than 2024-10-02:
  android_debug
  android_debug_arm64
  android_debug_unopt
  android_debug_unopt_arm64_ide
  host_debug_unopt_arm64_ide
  ios_debug
  ios_debug_sim_unopt_arm64
  ios_debug_unopt
  ios_debug_unopt_arm64
  license_script_output
  wasm_release
  web_tests
Run without --dry-run to reclaim 2.79gb.
```

```sh
flutter % et cleanup          
Checking /Users/matanl/Developer/engine/src/out...
Deleted 12 output directories and reclaimed 2.79gb.
```

### CODE REVIEW

File: tools/engine_tool/BUILD.gn [hunk 1]
```diff
@@ -40,6 +41,10 @@ dart_test("build_command_test") {
   main_dart = "test/commands/build_command_test.dart"
 }
 
+dart_test("cleanup_command_test") {
```
Comment (jtmcdole) [inline tools/engine_tool/BUILD.gn:44]: 🚀

File: tools/engine_tool/lib/src/commands/cleanup_command.dart [hunk 2]
```diff
@@ -0,0 +1,165 @@
...
+        int.parse(dateMatch.group(1)!),
+        int.parse(dateMatch.group(2)!),
+        int.parse(dateMatch.group(3)!),
+      );
+    }();
+
+    // Look at the directories in "out" for ones older than "since".
+    environment.logger.status('Checking ${environment.engine.outDir.path}...');
+    final toDelete = [
+      for (final entity in environment.engine.outDir.listSync())
```
Comment (jtmcdole) [inline tools/engine_tool/lib/src/commands/cleanup_command.dart:72]: we can't do `await for` in the list expansion can we?

File: tools/engine_tool/lib/src/commands/cleanup_command.dart [hunk 3]
```diff
@@ -0,0 +1,165 @@
...
+    type = _FileSize.gigabytes;
+    bytes = bytes / 1024;
+  }
+  return '${bytes.toStringAsFixed(2)}${type.suffix}';
+}
+
+enum _FileSize {
+  bytes('bytes'),
+  kilobytes('kb'),
+  megabytes('mb'),
```
Comment (jtmcdole) [inline tools/engine_tool/lib/src/commands/cleanup_command.dart:160]: the most NIT of NITs; I think these need to be KB for 1024, and kb for 1000.

**Review by jtmcdole — APPROVED** [review]

File: tools/engine_tool/lib/src/commands/cleanup_command.dart (hunk 2 above)
Comment (matanlurey) [reply tools/engine_tool/lib/src/commands/cleanup_command.dart]: We can (and do now)!

File: tools/engine_tool/lib/src/commands/cleanup_command.dart (hunk 3 above)
Comment (matanlurey) [reply tools/engine_tool/lib/src/commands/cleanup_command.dart]: Works for me.

**Comment by chinmaygarde** [comment]
The test failure is real. The following GN file needs to be re-formatted: `a/tools/engine_tool/BUILD.gn b/tools/engine_tool/BUILD.gn`

**Pull request merged**